iam.tf

```
@@ -81,6 +80,43 @@ resource "azurerm_role_assignment" "castai_additional_resource_groups" {
   scope              = each.key
 }
 
+# Add secondary role assignments to avoid cyclic dependency with cluster object.
+resource "azurerm_role_assignment" "castai_resource_group_2" {
```

```suggestion
resource "azurerm_role_assignment" "castai_resource_group_names" {
```

maybe something like that?

---

done